src/wallet/tx_builder.rs

```diff
@@ -483,9 +514,11 @@ impl<D: Database> TxBuilder<D, DefaultCoinSelectionAlgorithm, BumpFee> {
     /// Fails if the transaction has more than one outputs.
     ///
     /// [`add_utxo`]: Self::add_utxo
-    /// [`utxos`]: Self::utxos
-    pub fn maintain_single_recipient(mut self) -> Self {
-        self.single_recipient = Some(Script::default());
+    pub fn maintain_single_recipient(&mut self) -> &mut Self {
+        let mut recipients = self.params.recipients.drain(..).collect::<Vec<_>>();
+        assert_eq!(recipients.len(), 1, "maintain_single_recipient must not be called while bumping a transactions with more than one output");
```

Comment:
I remember we had a brief discussion about making the methods return `Result`s. Is there any reason why you didn't do that here?

Reply:
Good point. A result would be better in this case instead of the panic -- it would be weird to return an bdk::Error here though since there is only one very particular thing that can go wrong. Perhaps I can just return a `Error::Generic` for now and we can refine it later?

Reply:
Yeah, that's good enough for now. We have to refactor the whole error-handling stuff anyways, so we'll just update it once we do that

Reply:
It turns out we had an Error for this already that used to be thrown by the now non-existent `bump_fee` function so I just used that.